graphix/gflow.py

```diff
@@ -432,17 +456,109 @@ def get_layers(l_k):
     -------
     d: int
         minimum depth of graph
-    layers: dict of lists
+    layers: dict of set
         components of each layer
     """
     d = get_min_depth(l_k)
-    layers = {k: [] for k in range(d + 1)}
+    layers = {k: set() for k in range(d + 1)}
     for i in l_k.keys():
-        layers[l_k[i]].append(i)
+        layers[l_k[i]] |= {i}
     return d, layers
 
 
-def get_adjacency_matrix(graph):
+def get_dependence_flow(
+    inputs: set[int],
+    flow: dict[int, set[int]],
+    odd_flow: dict[int, set[int]],
+) -> dict[int, set[int]]:
+    """Get dependence flow from flow.
+
+    Parameters
+    ----------
+    inputs: set[int]
+        set of input nodes
+    flow: dict[int, set]
+        flow function. flow[i] is the set of qubits to be corrected for the measurement of qubit i.
+    odd_flow: dict[int, set]
+        odd flow neighbors of each node.
+        odd_flow[i] is the set of odd neighbors of f(i).
+
+    Returns
+    -------
+    dependence_flow: dict[int, set]
+        dependence flow function. dependence_flow[i] is the set of qubits to be corrected for the measurement of qubit i.
+    """
+    dependence_flow = {input: set() for input in inputs}
+    # concatenate flow and odd_flow
+    combined_flow = dict()
+    for node, corrections in flow.items():
+        combined_flow[node] = corrections | odd_flow[node]
+    for node, corrections in combined_flow.items():
+        for correction in corrections:
+            if correction not in dependence_flow.keys():
+                dependence_flow[correction] = set()
+            dependence_flow[correction] |= {node}
+    return dependence_flow
+
+
+def get_layers_from_flow(
+    flow: dict[int, set], odd_flow: dict[int, set], inputs: set[int], outputs: set[int]
+) -> tuple[dict[int, set], int]:
+    """Get layers from flow (incl. gflow).
+
+    Parameters
+    ----------
+    flow: dict[int, set]
+        flow function. flow[i] is the set of qubits to be corrected for the measurement of qubit i.
+    odd_flow: dict[int, set]
+        odd flow neighbors of each node.
```

Comment:
so it's different from odd neighbors right? I would recommend noting the definition (using math environment) to docstring.

Reply:
Sorry, it's just odd neighbors of (g)flow. Odd(f(node)) or Odd(g(node))